ocs_ci/helpers/vdbench_helpers.py

```diff
@@ -0,0 +1,696 @@
+        lines = logs.split("\n")
+        for line in lines:
+            line = line.strip()
+
+            # Parse summary statistics
+            if "avg_rate" in line and "resp" in line:
+                # Example line: "avg_rate=1234.5 resp=5.67"
+                parts = line.split()
+                for part in parts:
+                    if "=" in part:
+                        key, value = part.split("=", 1)
+                        try:
+                            metrics[key] = float(value)
+                        except ValueError:
+                            metrics[key] = value
+
+            # Parse other relevant metrics
+            if "Total" in line and "rate=" in line:
+                # Parse total statistics
+                if "rate=" in line:
+                    rate_part = line.split("rate=")[1].split()[0]
+                    try:
+                        metrics["total_rate"] = float(rate_part)
+                    except ValueError:
+                        pass
+
+        return metrics if metrics else None
+
+    except AttributeError as e:
+        log.error(f"Invalid input type for logs. Expected string. Error: {e}")
+        return None
+
+
+def monitor_vdbench_workload(workload, interval=30, duration=300):
+    """
+    Monitor a running Vdbench workload and collect metrics.
+
+    Args:
+        workload: VdbenchWorkload instance
+        interval: Monitoring interval in seconds
+        duration: Total monitoring duration in seconds
+
+    Returns:
+        list: List of metric snapshots
+    """
+    metrics_history = []
+    start_time = time.time()
+
+    while (time.time() - start_time) < duration:
+        try:
+            # Get current status
+            status = workload.get_workload_status()
+
+            # Get logs and parse metrics
+            logs_dict = get_vdbench_logs(workload.namespace, workload.deployment_name)
+
+            snapshot = {"timestamp": time.time(), "status": status, "metrics": {}}
+
+            # Parse metrics from each pod
+            for pod_name, logs in logs_dict.items():
+                try:
+                    pod_metrics = parse_vdbench_output(logs)
+                    if pod_metrics:
+                        snapshot["metrics"][pod_name] = pod_metrics
+                except (ValueError, KeyError, TypeError) as parse_error:
+                    log.warning(
+                        f"Failed to parse metrics for pod {pod_name}: {parse_error}"
+                    )
+
+            metrics_history.append(snapshot)
+            log.info(f"Collected metrics snapshot {len(metrics_history)}")
+            time.sleep(interval)
+
+        except (AttributeError, RuntimeError) as workload_error:
+            log.error(f"Workload error during monitoring: {workload_error}")
+            break
+        except (OSError, TimeoutError) as system_error:
+            log.error(f"System/log access error: {system_error}")
+            break
+        except Exception as e:
```

Comment:
are you sure you want to catch any exception? it could mask un expected issues

Reply:
YEs Not needed, I have removed it